aptos-move/framework/aptos-framework/sources/vesting.move

```diff
@@ -401,18 +401,17 @@ module aptos_framework::vesting {
             return shareholder_or_beneficiary
         };
         let vesting_contract = borrow_global<VestingContract>(vesting_contract_address);
-        let i = 0;
-        let len = vector::length(shareholders);
-        while (i < len) {
-            let shareholder = *vector::borrow(shareholders, i);
-            // This will still return the shareholder if shareholder == beneficiary.
-            if (shareholder_or_beneficiary == get_beneficiary(vesting_contract, shareholder)) {
-                return shareholder
-            };
-            i = i + 1;
-        };
+        let result = @0x0;
+        vector::any(shareholders, |shareholder| {
```

Comment:
I wonder if we can add a find() function in vector that returns the ref to the first element that matches a predicate. I've seen this use case in other places updated in this PR as well

Reply:
I used consider this but didn't come up with a good solution. My rationales:
1. find() cannot return &Element because the element that matches the predicate might not exist.
2. find() cannot return Element due to ownership.
3. We cannot use `Option` because its implement is based on `vector`. I think it doesn't make sense though.
4. If we return the index of the element, we still need to compare with the length to check if we find something then `borrow` that element with the index. In this case `any` is even more convenient...

Reply:
I see that makes sense